File: References/Java/SphrLibrary-Java/lib/src/main/java/sphrlibrary/entities/openmhealth/TimeInterval.java
package sphrlibrary.entities.openmhealth;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;

/**
 * 時間間隔
 */
@Data
@NoArgsConstructor
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
public class TimeInterval {

    /**
     * 開始日時
     */
    @NonNull
    private String startDateTime = "";

    /**
     * 終了日時
     */
    @NonNull
    private String endDateTime = "";
}

File: SphrLibrary/Entities/OpenmHealth/TimeInterval.cs
using System.Runtime.Serialization;

namespace SphrLibrary.Entities.OpenmHealth
{
    /// <summary>
    /// Open mHealth TimeInterval リソースを表します。
    /// </summary>
    [DataContract()]
    [Serializable()]
    public class TimeInterval
    {
        [DataMember()]
        public string start_date_time = string.Empty;
        [DataMember()]
        public string end_date_time = string.Empty;

        public TimeInterval() { }
    }
}
